### DISCUSSION TITLE
Availability chart empty

### DISCUSSION THREAD

**Discussion by mirkospasic** [berezovskyi/sparqles#9] [Q&A] [answered]
Hello! I have an issue with the availability chart on the main page. Can you help me with it? It is empty, and it says: "No data available". It seems that I don't have the amonth collection in my MongoDB instance...

**Comment by mirkospasic** [comment]
![Screenshot(3)](https://user-images.githubusercontent.com/5012363/205625960-0d36c265-5573-434b-814c-a995573062d0.png)

**Reply by berezovskyi** [reply]
Hi @mirkospasic! My apologies for missing this issue completely and, of course, for not getting back to you earlier. I recently made an effort (https://github.com/berezovskyi/sparqles/releases/tag/v0.3.0) to update all security issues in the tool and get it running again (partly because the TU Wien installation has been down for almost a whole month again). It is now running well except for the Availability chart you mentioned (though it is not public yet - but all the code updates are pushed and available):

![sparqles sv berezovskyi me_](https://github.com/user-attachments/assets/3d66af79-a85b-42e3-8561-fed791997335)


The chart in question is supposed to show the evolution of the availability of all endpoints on a month-to-month basis. The issue you reported is indeed a bug, because the code to update the necessary table in the DB ("collection" in the MongoDB parlance) is not wired up to run as part of the system. I filed #57 to track the work on this. In the meantime, I will let the instance to collect some data.

If you have any SPARQL endpoints to propose (or better yet, places to scrape lists of endpoints), please tell me in #14

**Reply by berezovskyi** [reply]
Ha, I just noticed that you fixed many issues in https://github.com/OpenLinkSoftware/sparqles/commits/main/. I will try to cherry-pick most of your changes. Thanks Mirko for making your changes open as well!

**Reply by berezovskyi** [reply, marked as answer]
@mirkospasic thanks for the fixes in your fork of our fork - I tried to pull most of them in https://github.com/berezovskyi/sparqles/releases/tag/v0.4.0
